internal/provider/network_connection_resource.go

```diff
@@ -0,0 +1,172 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func NewNetworkConnectionResource() resource.Resource {
+	return &NetworkConnectionResource{}
+}
+
+type NetworkConnectionResource struct {
+	sdk                      *dcsdk.SDK
+	networkConnectionService *dcgennet.NetworkConnectionServiceClient
+}
+
+func (r *NetworkConnectionResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_network_connection"
+}
+
+func (r *NetworkConnectionResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = networkConnectionResourceSchema
+}
+
+func (r *NetworkConnectionResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	sdk, ok := req.ProviderData.(*dcsdk.SDK)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *dcsdk.SDK, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.sdk = sdk
+	r.networkConnectionService = r.sdk.Network().NetworkConnection()
+}
+
+func (r *NetworkConnectionResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *NetworkConnectionModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	createReq := &network.CreateNetworkConnectionRequest{
+		NetworkId:   data.NetworkID.ValueString(),
+		Params:      nil,
+		Description: data.Description.ValueString(),
+	}
+
+	switch {
+	case data.AWS != nil:
+		aws := &network.CreateAWSNetworkConnectionRequest{}
+		createReq.Params = &network.CreateNetworkConnectionRequest_Aws{
+			Aws: aws,
+		}
+		switch {
+		case data.AWS.Peering != nil:
+			aws.Type = &network.CreateAWSNetworkConnectionRequest_Peering{
+				Peering: &network.CreateAWSNetworkConnectionPeeringRequest{
+					VpcId:         data.AWS.Peering.VPCID.ValueString(),
+					AccountId:     data.AWS.Peering.AccountID.ValueString(),
+					RegionId:      data.AWS.Peering.RegionID.ValueString(),
+					Ipv4CidrBlock: data.AWS.Peering.IPv4CIDRBlock.ValueString(),
+					Ipv6CidrBlock: data.AWS.Peering.IPv6CIDRBlock.ValueString(),
+				},
+			}
+		default:
+			resp.Diagnostics.AddError("misconfiguration", "\"aws.peering\" must be specified")
+			return
+		}
+	case data.Google != nil:
+		createReq.Params = &network.CreateNetworkConnectionRequest_Google{
+			Google: &network.CreateGoogleNetworkConnectionRequest{
+				Name:           data.Google.Name.ValueString(),
+				PeerNetworkUrl: data.Google.PeerNetworkURL.ValueString(),
+			},
+		}
+	default:
+		resp.Diagnostics.AddError("misconfiguration", "at least one of \"aws\" or \"google\" must be specified")
```

Comment:
You already have this validation in schema.
No need to write it twice.

Reply:
I can remove `default` statement here. But then code will be unclean IMHO

Reply:
OK, let's leave it.
Agree with you.